tests/integration/device_test.sh

```diff
@@ -4,87 +4,261 @@ include './tests/unit/utils.sh'
 include './tests/integration/utils.sh'
 include './src/device_info.sh'
 
+declare -gA results
+
 function oneTimeSetUp()
 {
   # The VERBOSE variable is set and exported in the run_tests.sh script based
   # on the command-line options provided by the user. It controls the verbosity
   # of the output during the test runs.
   setup_container_environment "$VERBOSE" 'device'
+
+  # Ensure kw_device_output is called once to populate the results array
+  kw_device_output
 }
 
-# test if `kw device --local` in the container matches the host
-function device_info_test_helper()
+# Test if `kw device --local` in the container matches the host
+function kw_device_output()
 {
-  local distro="${1}"
+  local distro
   local container
-  local buffer
+  local container_output
   local filter
   local filter_regex
-  local fs_type
-  local fs_size
   local output
+  local output_filter=(
+    'Distribution base' 'Distribution version' 'Desktop environments' # Fields to filter out
+  )
+
+  for distro in "${DISTROS[@]}"; do
+    container="kw-${distro}"
+
+    # Get the output of the `kw device --local` command from the container
+    container_output=$(container_exec "$container" 'kw device --local')
+    # Construct a regex by joining the filter items with a | separator
+    filter_regex=$(printf '%s|' "${output_filter[@]}" | sed 's/|$//')
+    # The actual perl regex.
+    filter_regex="(${filter_regex}):"
+    # Get the output, after applying filter.
+    output=$(printf '%s' "${container_output}" | grep --invert-match --perl-regexp "$filter_regex")
+    # Store the result in the array
+    results["$distro"]="$output"
+  done
+}
+
+function test_kw_device_local_ram()
+{
+  local distro
+  local container
+  local mem_total_kb
   local expected_output
+  local output
 
-  # buffer with all information.
-  container="kw-${distro}"
-  buffer=$(container_exec "${container}" 'kw device --local')
+  for distro in "${!results[@]}"; do
+    container="kw-${distro}"
+    output="${results[$distro]}"
+    # Get the RAM value in kilobytes from the container
+    mem_total_kb=$(container_exec "$container" "awk '/MemTotal/ {print \$2}' /proc/meminfo")
+    # Convert the RAM value to IEC format
+    expected_output=$(container_exec "$container" "numfmt --from=si --to=iec ${mem_total_kb}K")
+    # Extract RAM value from the output
+    output=$(printf "%s" "$output" | grep "Total RAM" | awk '{print $3}')
```

Comment:
awk in fine, but maybe a simple `cut -d` can get the job done and it is very easy to understand.